### DISCUSSION TITLE
go install golangci-lint failed. "ld: framework not found CoreFoundation"

### DISCUSSION THREAD

**Discussion by AGou-ops** [golangci/golangci-lint#2429] [Q&A]
error:
```bash
> go version
go version go1.17.5 darwin/arm64
> go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.42.1
# github.com/golangci/golangci-lint/cmd/golangci-lint
/usr/local/go/pkg/tool/darwin_arm64/link: running clang failed: exit status 1
ld: framework not found CoreFoundation
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
os: MacOS 12.1 Monterey

what I have tried:

- reinstall golang(homebrew and offical pkg)

How to fix it? Thx.

**Comment by butuzov** [comment]
Try to reinstall Xcode
```
xcode-select --install
```

**Reply by AGou-ops** [reply]
I already installed xcode.
And I tried to reinstall xcode but it still doesn't work.

**Reply by butuzov** [reply]
Then check if [CoreFoundation](https://developer.apple.com/documentation/corefoundation) needs to be installed implicitly. It supposes to be installed with xcode (check `/Library/Frameworks/`).

**Reply by AGou-ops** [reply]
Thanks for your reply.

no CoreFoundation in `/Library/Frameworks/`:
```bash
> ll /Library/Frameworks/
  rwxr-xr-x   1   root   wheel    582 KiB   Sat Nov 20 14:32:51 2021    679F
  rwxr-xr-x   5   root   wheel    160 B     Wed Dec  8 07:39:27 2021    CoreRepairCore.framework/
  rwxr-xr-x   5   root   wheel    160 B     Wed Dec  8 07:39:27 2021    CoreRepairKit.framework/
  rwxr-xr-x   1   root   wheel     50 B     Wed Dec  8 07:39:27 2021    iTunesLibrary.framework  ⇒ /System/Library/Frameworks/iTunesLibrary.framework
```

Should I install `CoreFoundation`? I found a source code repo [Github CoreFoundation](https://github.com/macmade/CoreFoundation), but I don't know how to install it.😅

**Reply by dowdt** [reply]
@AGou-ops How did you install it?
